hack/release.sh

```diff
@@ -64,6 +64,13 @@ cp "${REPO_ROOT_DIR}/third_party/config/build/release.yaml" "${BUILD_YAML}"
 
 echo "Building Knative Serving"
 ko resolve ${KO_FLAGS} -f config/ > "${SERVING_YAML}"
+if [[ -n "${RELEASE_VERSION}" ]]; then
+  # Add the configmap including release information
+  COMMIT_ID=$(cat .git/HEAD)
```

Comment:
As outlined elsewhere, I think I'd rather see us:
1. Ship a library where we publish `${KO_DATA_PATH}/HEAD` as a prometheus metric, and
2. Link `.git/HEAD` to the `./cmd/*/kodata/` directory of each binary (and link the above).

This information should be strictly stronger as it's versioned with each binary vs. an orthogonal configmap.